chainercv/links/model/feature_predictor.py

```diff
@@ -0,0 +1,171 @@
+from chainercv.transforms import resize
+from chainercv.transforms import scale
+from chainercv.transforms import ten_crop
+
+
+class FeaturePredictor(chainer.Chain):
+
+    """Wrapper that adds a prediction method to a feature extraction model.
+
+    The :meth:`predict` takes three steps to make a prediction.
+
+    1. Preprocess input images
+    2. Forward the preprocessed images to the network
+    3. Average features in the case when more than one crops are extracted.
+
+    Example:
+
+        >>> from chainercv.links import VGG16
+        >>> from chainercv.links import FeaturePredictor
+        >>> base_model = VGG16()
+        >>> model = FeaturePredictor(base_model, 224, 256)
+        >>> prob = model.predict([img])
+        # Predicting multiple features
+        >>> model.extractor.feature_names = ['conv5_3', 'fc7']
+        >>> conv5_3, fc7 = model.predict([img])
+
+    When :obj:`self.crop == 'center'`, :meth:`predict` extracts features from
+    the center crop of the input images.
+    When :obj:`self.crop == '10'`, :meth:`predict` extracts features from
+    patches that are ten-cropped from the input images.
+
+    When extracting more than one crops from an image, the output of
+    :meth:`predict` returns the average of the features computed from the
+    crops.
+
+    Args:
+        extractor: A feature extraction model. This is a callable chain
+            that takes a batch of images and returns a variable or a
+            tuple of variables.
+        crop_size (int or tuple): The height and the width of an image after
+            cropping in preprocessing.
+            If this is an integer, the image is cropped to
+            :math:`(crop_size, crop_size)`.
+        scale_size (int or tuple): If :obj:`scale_size` is :obj:`None`,
+            neither scaling nor resizing is conducted during preprocessing.
+            This is the default behavior.
+            If this is an integer, an image is resized so that the length of
+            the shorter edge is equal to :obj:`scale_size`. If this is a tuple
+            :obj:`(height, width)`, the image is resized to
+            :math:`(height, width)`.
+        crop ({'center', '10'}): Determines the style of cropping.
+        mean (numpy.ndarray): A mean value. If this is :obj:`None`,
+            :obj:`extractor.mean` is used as the mean value.
+
+    """
+
+    def __init__(self, extractor,
+                 crop_size, scale_size=None,
+                 crop='center', mean=None):
+        super(FeaturePredictor, self).__init__()
+        self.scale_size = scale_size
+        if isinstance(crop_size, int):
+            crop_size = (crop_size, crop_size)
+        self.crop_size = crop_size
+        self.crop = crop
+        self._mean = mean
+
+        with self.init_scope():
+            self.extractor = extractor
+
+    @property
+    def mean(self):
+        if self._mean is None:
+            return self.extractor.mean
+        else:
+            return self._mean
```

Comment:
How about checking existence of `mean` in `__init__`?
```python
    def __init__(...):
        ...
        if mean is None:
            self.mean = self.extractor.mean
        else:
            self.mean = mean
```
I prefer this version because it can notify the absence of `mean` before `predict`.

In the case of current version:
```python
extractor = ExtractorWithoutMean()
predictor = FeaturePredictor(extractor, crop_size=224)  # mean should be specified but no errors are shown here
...
predictor.predict(imgs)  # An exception will be raised
```